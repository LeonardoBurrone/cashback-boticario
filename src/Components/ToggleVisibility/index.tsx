import { IconButton, InputAdornment } from '@material-ui/core';
import { VisibilityOffOutlined, VisibilityOutlined } from '@material-ui/icons';
import React from 'react';

export type Props = {
  show: boolean;
  toggle: () => void;
};

export const ToggleVisibility: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <InputAdornment position={'end'}>
      <IconButton aria-label={'toggle visibility'} edge={'end'} onClick={props.toggle}>
        {props.show ? <VisibilityOutlined /> : <VisibilityOffOutlined />}
      </IconButton>
    </InputAdornment>
  );
};

export default ToggleVisibility;
