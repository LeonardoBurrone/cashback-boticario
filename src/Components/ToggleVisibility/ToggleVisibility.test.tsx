import { IconButton } from '@material-ui/core';
import { VisibilityOffOutlined, VisibilityOutlined } from '@material-ui/icons';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import ToggleVisibility, { Props } from './';

describe('Test component ToggleVisibility', () => {
  const toggleSpy = jest.fn(() => ({}));

  const props: Props = {
    show: false,
    toggle: toggleSpy
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<ToggleVisibility {...props} />);
  });

  it('Check if component rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Validate button icon when button is turned off', () => {
    expect(wrapper.find(VisibilityOffOutlined).exists()).toBe(true);
  });

  it('Validate button icon when button is turned on', () => {
    wrapper.setProps({
      show: true
    });
    expect(wrapper.find(VisibilityOutlined).exists()).toBe(true);
  });

  it('Check if toggle function is called when pression the button', () => {
    const button: any = wrapper.find(IconButton).props();
    button.onClick();

    expect(toggleSpy).toBeCalled();
  });
});
