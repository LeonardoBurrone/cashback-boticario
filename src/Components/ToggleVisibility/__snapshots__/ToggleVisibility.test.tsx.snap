// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component ToggleVisibility Check if component rendered correctly 1`] = `
<WithStyles(ForwardRef(InputAdornment))
  position="end"
>
  <WithStyles(ForwardRef(IconButton))
    aria-label="toggle visibility"
    edge="end"
    onClick={[MockFunction]}
  >
    <Memo(ForwardRef(VisibilityOffOutlinedIcon)) />
  </WithStyles(ForwardRef(IconButton))>
</WithStyles(ForwardRef(InputAdornment))>
`;
